### PR TITLE
Add support for importing patterns via URL inside services

### PR DESCRIPTION
**Pull request by Revolyssup**
**Description**

This PR adds support for allowing composable designs.
- Designs can externalise internal fields to make them configurable by any higher order design that imports them.
- Currently designs can only be imported by URL
- `vars` field allows for having default values which can be overriden by calling design

**Notes for Reviewers**

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by leecalcote** [comment]
New error codes?

**Comment by Revolyssup** [comment]
> New error codes?

Yups will add it. Created the PR before the demo. Will add the errrors

File: models/pattern/stages/import.go [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+}
+func process(imp *servicestack, nonimp *servicestack, vars map[string]interface{}) error {
+	for !imp.isEmpty() {
+		sw := imp.pop()
+		url, ok := matchImportPattern(sw.svc.Type)
+		if !ok {
+			nonimp.push(sw)
+			continue
+		}
+		p, err := getPatternFromLocation(url)
```
Comment (sudo-NithishKarthik) [inline models/pattern/stages/import.go:126]: wouldn't we want to validate this pattern file before we start processing it 🤔 ? @Revolyssup

**Review by sudo-NithishKarthik — DISMISSED** [review]
LGTM 🎉

**Comment by Revolyssup** [comment]
@leecalcote This PR is tested and ready to merge. After this is merged, no older pattern would break.
I will modify existing patterns to be more generic

**Review by sudo-NithishKarthik — APPROVED** [review]
LGTM 🎉

**Comment by leecalcote** [comment]
The abbreviated variable names make legibility more of a stain than is ideal. 

Imp
Nonimp 
Swc
Act
Sip
Sap
...

**Comment by leecalcote** [comment]
A new stage is being added in this PR. This is a moderately sized change in behavior. Will you include a link to the updated documentation/design spec that includes a description of the stage and how it fits into the overall engine?

**Pull request merged**